persistent/test/main.hs

```diff
@@ -1,891 +1,16 @@
-            describe "idTable" $ do
-                let EntityDef {..} = idTable
-                it "has no extra blocks" $ do
-                    entityExtra `shouldBe` mempty
-                it "has the right name" $ do
-                    entityHaskell `shouldBe` EntityNameHS "IdTable"
-                it "has the right fields" $ do
-                    map fieldHaskell entityFields `shouldMatchList`
-                        [ FieldNameHS "name"
-                        ]
-            describe "lowerCaseTable" $ do
-                let EntityDef {..} = lowerCaseTable
-                it "has the right name" $ do
-                    entityHaskell `shouldBe` EntityNameHS "LowerCaseTable"
-                it "has the right fields" $ do
-                    map fieldHaskell entityFields `shouldMatchList`
-                        [ FieldNameHS "fullName"
-                        ]
-                it "has ExtraBlock" $ do
-                    Map.lookup "ExtraBlock" entityExtra
-                        `shouldBe` Just
-                            [ ["foo", "bar"]
-                            , ["baz"]
-                            , ["bin"]
-                            ]
-                it "has ExtraBlock2" $ do
-                    Map.lookup "ExtraBlock2" entityExtra
-                        `shouldBe` Just
-                            [ ["something"]
-                            ]
-
-    describe "fromPersistValue" $
-        describe "UTCTime" $
-            it "works with format" $
-                fromPersistValue (PersistText "2018-02-27 10:49:42.123")
-                    `shouldBe` Right (UTCTime (fromGregorian 2018 02 27) (timeOfDayToTime (TimeOfDay 10 49 42.123)))
-
-    describe "PersistValue" $ do
-        describe "Aeson" $ do
-            let
-                testPrefix constr prefixChar bytes =
-                    takePrefix (toJSON (constr (BS8.pack bytes)))
-                    ===
-                    String (T.singleton prefixChar)
-                roundTrip constr bytes =
-                    fromJSON (toJSON (constr (BS8.pack bytes)))
-                    ===
-                    Data.Aeson.Success (constr (BS8.pack bytes))
-                subject constr prefixChar = do
-                    prop ("encodes with a " ++ [prefixChar] ++ " prefix") $
-                        testPrefix constr prefixChar
-                    prop "Round Trips" $
-                        roundTrip constr
-
-            describe "PersistDbSpecific" $ do
-                subject PersistDbSpecific 'p'
-            describe "PersistLiteral" $ do
-                subject PersistLiteral 'l'
-            describe "PersistLiteralEscaped" $ do
-                subject PersistLiteralEscaped 'e'
-
-takePrefix :: Value -> Value
-takePrefix (String a) = String (T.take 1 a)
-takePrefix a = a
-
-arbitraryWhiteSpaceChar :: Gen Char
-arbitraryWhiteSpaceChar =
-  oneof $ pure <$> [' ', '\t', '\n', '\r']
+    describe "Database" $ describe "Persist" $ do
```

Comment:
I'm not too sure what has happened here, but it seems like the merge of my PR has resulted in some of the changes coming in but not others.

I did an initial merge last night (which was before you merged your QQ changes in), and following you merging in your QQ PR, merged this branch back into mine again, but it seems like the changes from the first commit have come through here (https://github.com/yesodweb/persistent/pull/1261/commits/5a44a6d358db5e6df4f891e2fc90764784267f64) but not the second merge (https://github.com/yesodweb/persistent/pull/1261/commits/8d4e5b25cb32df52db887570d0eb35e49400b873). You may have clocked this already but I just wanted to point it out in case you hadn't.

Note the conflicts on this PR still, and the lack of a `KindEntities.spec` call in these tests.

You might be able to just cherry pick `8d4e5b25cb32df52db887570d0eb35e49400b873` on this branch, then push that up? :shrug: I'm not too sure what has happened here

Reply:
no worries! I got it fixed up